Sources/StreamChat/StateLayer/ReactionList.swift

```diff
@@ -0,0 +1,87 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// An object which represents a list of `ChatMessageReaction` for the specified query.
+@available(iOS 13.0, *)
+public final class ReactionList {
+    private let query: ReactionListQuery
+    private let reactionListUpdater: ReactionListUpdater
+    private let stateBuilder: StateBuilder<ReactionListState>
+    
+    init(query: ReactionListQuery, client: ChatClient, environment: Environment = .init()) {
+        self.query = query
+        reactionListUpdater = environment.reactionListUpdaterBuilder(
+            client.databaseContainer,
+            client.apiClient
+        )
+        stateBuilder = StateBuilder {
+            environment.stateBuilder(
+                query,
+                client.databaseContainer
+            )
+        }
+    }
+    
+    /// An observable object representing the current state of the reaction list.
+    @MainActor public lazy var state: ReactionListState = stateBuilder.build()
+    
+    /// Fetches the most recent state from the server and updates the local store.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    public func get() async throws {
+        let pagination = Pagination(pageSize: 25)
+        try await loadReactions(with: pagination)
+    }
+    
+    /// Loads reactions for the specified pagination parameters and updates ``ReactionListState/reactions``.
+    ///
+    /// - Parameter pagination: The pagination configuration which includes a limit and an offset or a cursor.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    /// - Returns: An array of message reaction for the pagination.
+    @discardableResult public func loadReactions(with pagination: Pagination) async throws -> [ChatMessageReaction] {
+        let query = query.withPagination(pagination)
+        return try await reactionListUpdater.loadReactions(query: query)
+    }
+    
+    /// Loads more message reactions and updates ``MemberListState/members``.
+    ///
+    /// - Parameters
+    ///   - limit: The limit for the page size. The default limit is 25.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    /// - Returns: An array of message reactions.
+    @discardableResult public func loadMoreReactions(limit: Int? = nil) async throws -> [ChatMessageReaction] {
```

Comment:
Maybe there is no point on limit being nil? 🤔 
```suggestion
    @discardableResult public func loadMoreReactions(limit: Int = 25) async throws -> [ChatMessageReaction] {
```

Reply:
Currently it just matches with other methods. But it is a fair point, optional Int looks a bit funny. If we go for it, we should change it in other types as well (separate PR). I am up for it changing it everywhere to non-optionals.

Reply:
I am gonna create another PR which changes all the call sites.